graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -226,14 +231,16 @@ def _validate_renamings(
         renamings: maps original type name to renamed name or None (for type suppression). Any name
                    not in the dict will be unchanged
         query_type: name of the query type, e.g. 'RootSchemaQuery'
+        scalars: set of all scalars used in the schema, including user defined scalars and used
+                 builtin scalars, excluding unused builtins
 
     Raises:
         - CascadingSuppressionError if a type suppression would require further suppressions
         - NotImplementedError if renamings attempts to suppress an enum, an interface, or a type
           implementing an interface
     """
     _check_for_cascading_type_suppression(schema_ast, renamings, query_type)
-    _ensure_no_unsupported_suppression(schema_ast, renamings)
+    _check_for_unsupported_operation(schema_ast, renamings, scalars)
```

Comment:
Consider using a plural form, `operations`. Also, I kind of prefer `ensure` since it more clearly communicates the outcome if an unsupported operation is found. When I see `ensure` I definitely think "error will be raised if anything is found", when I see `check` I may very well think "it'll return a value telling me what it found".